itests/kit/mir_tools.go

```diff
@@ -219,16 +223,19 @@ func ChainHeightCheckWithFaultyNodes(ctx context.Context, blocks int, nodes []*T
 	}
 
 	for i, fn := range faultyNodes {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
 		ts, err := fn.FullNode.ChainHead(ctx)
 		if err != nil {
 			return err
 		}
```

Comment:
What about making this an independent function (I guess it can live in the Mir package for now as it is the only one using it.

Pseudocode: 
```suggestion
func ChainHeadWithCtx(){
		if ctx.Err() != nil {
			return ctx.Err()
		}
		ts, err := fn.FullNode.ChainHead(ctx)
		if err != nil {
			return err
		}
}
```

Reply:
That does not work because we access ChainHead via API. To do that, we need to extend API.

Reply:
Fixed